src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -71,7 +71,7 @@ QgsMeshLayer::QgsMeshLayer( const QString &meshLayerPath,
   resetDatasetGroupTreeItem();
   setLegend( QgsMapLayerLegend::defaultMeshLegend( this ) );
 
-  if ( isValid() && options.loadDefaultStyle )
+  if ( isValid() )
```

Comment:
is this change required to fix the issue? It looks wrong to me -- if we aren't specifying that flag, then it indicates that we'll be manually setting the renderer layer later (or don't care because we'll never render the layer)

Reply:
yes, that was required, but because there was a little mess with default style...
 Indeed, some confusion between 2 methods for default style since https://github.com/qgis/QGIS/commit/b48d7dcc9fd1e5d0a32ab87f37a17b41fa56d8dc I haven't seen.
Also, I didn't catch/remember the new logic for default style...
With the new commit, it should be OK now.